core/ui/prime/components/src/lib/autocomplete/autocomplete.component.ts

```diff
@@ -3,43 +3,123 @@ import { NG_VALUE_ACCESSOR } from "@angular/forms";
 import { InputBaseComponent } from "../input-base/input-base.component";
 
 @Component({
-    selector: "lens-autocomplete",
-    templateUrl: "autocomplete.component.html",
-    providers: [
-        { provide: NG_VALUE_ACCESSOR, useExisting: forwardRef(() => AutoCompleteComponent), multi: true }
-    ]
+  selector: "lens-autocomplete",
+  templateUrl: "autocomplete.component.html",
+  providers: [
+    { provide: NG_VALUE_ACCESSOR, useExisting: forwardRef(() => AutoCompleteComponent), multi: true }
+  ]
 })
 export class AutoCompleteComponent extends InputBaseComponent implements OnInit {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    @Input() public options: { key: any, value: string }[] = [];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  @Input() public options: { key: any, value: string }[] = [];
+  @Input() public multiple = false;
+  @Input() public forceSelection = true;
+  @Input() public dropdown = true;
+  @Input() public modeType: "default" | "tags" = "default";
+  @Input() public separator = ""; // separator character to add an item when pressed in addition to the enter key.
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    public filteredOptions: { key: any, value: string }[] = [];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public filteredOptions: { key: any, value: string }[] = [];
 
-    public ngOnInit(): void {
-        this.filteredOptions = this.options;
+  public ngOnInit(): void {
+    this.filteredOptions = this.options;
+    if (this.modeType === "tags") {
+      // multiple=true only works if you also set dropdown=false
+      this.multiple = true;
+      this.dropdown = false;
+      this.forceSelection = false;
     }
+  }
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    public onItemSelected(value: any): void {
-        this.value = value;
-        this.valueChanged();
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public onItemSelected(value: any): void {
+    if (this.modeType !== "tags") {
+      // TODO: [fix issue]: updating 'value' when modeType=tags, generates an error!
```

Comment:
Forgot this todo?

Reply:
it is fixed (autocomplete component is refactored) in the dageorge/masterdata-filter-tags branch